### ISSUE TITLE
`derivedFrom`: cannot provide `undefined` as `initialValue`

### ISSUE THREAD

**Issue by mattmoos** [ngxtension/ngxtension-platform#427]
I noticed that an initial value of `undefined` cannot be used in `derivedFrom`. I think this would be okay in itself because Angular is using `undefined` implicitly as initial value. But `derivedFrom` is setting `requireSync: true` when it detects no initial value, cf. 
https://github.com/ngxtension/ngxtension-platform/blob/658830c2fa6dc9eaf9f8e813c0509f9d8b5064bf/libs/ngxtension/derived-from/src/derived-from.ts#L112
and that does not work with async observables as sources

The problem seems to be here:
https://github.com/ngxtension/ngxtension-platform/blob/658830c2fa6dc9eaf9f8e813c0509f9d8b5064bf/libs/ngxtension/derived-from/src/derived-from.ts#L135
`derivedFrom` is checking `options?.initialValue !== undefined;`, thus an initial value of `undefined` gets lost.

I guess this could be fixed by using `Object.hasOwn(options, 'initialValue')` instead?

**Comment by eneajaho** [comment]
@mattmoos Can you provide a PR that adds a test case with your usecase (simplified) that breaks, and we can work on a fix or just use what you recommended (object.hasOwn).

**Comment by mattmoos** [comment]
Yes, just had a look at the existing test cases. I think I can provide a test.
